packages/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -236,6 +251,39 @@ class DescopeWc extends BaseDescopeWc {
       return;
     }
 
+    if (
+      action === RESPONSE_ACTIONS.loadForm &&
+      isChanged('samlIdpFormResponse')
+    ) {
+      if (!samlIdpFormResponse) {
+        this.loggerWrapper.error('Did not get saml form data to load');
+        return;
+      }
+
+      // Handle SAML IDP end of flow ("redirect like" by using html form with hidden params)
+
+      // create form tag from response string
+      const htmlFormTemp = document.createElement('div');
+      htmlFormTemp.innerHTML = samlIdpFormResponse;
+      const htmlForm = htmlFormTemp.querySelector('form');
+      document.body.appendChild(htmlForm);
+
+      // create script tags from response scripts
+      const htmlScripts = document.createElement('div');
+      htmlScripts.innerHTML = samlIdpFormResponse;
+      const scripts = Array.from(htmlScripts.getElementsByTagName('script'));
+      scripts.forEach((originalScript) => {
+        const script = document.createElement('script');
+        if (originalScript.src) {
+          script.src = originalScript.src;
+        } else {
+          script.textContent = originalScript.textContent;
+        }
+        document.body.appendChild(script);
+      });
+      return;
+    }
```

Comment:
Please extract all of this into a utility function

Reply:
@nirgur please review this part thoroughly, need to make sure it will work in v2 as well

Reply:
Extracted to function `injectSamlIdpForm`